src/main/webapp/profile.js

```diff
@@ -0,0 +1,191 @@
+// limitations under the License.
+
+
+/** Fetches the profile from the server and adds them to the DOM. */
+function loadProfile() {
+    fetch('/profile').then(response => response.json()).then((profile) => {
+
+        // initialize elements
+        const nameElement = document.getElementById("name");
+        const contactInfoElement = document.getElementById("contact-info");
+        const reviewListElement = document.getElementById("review-list");
+        // const projectListElement = document.getElementById("project-list");
+
+        // display profile
+        nameElement.innerText = profile.name;
+        console.log(profile.name);
+        contactInfoElement.innerHTML = profile.contactInfo.Email + "<br/>" + profile.contactInfo.Mobile;
+        console.log(profile.contactInfo.Email + "<br/>" + profile.contactInfo.Mobile);
+        contactInfoElement.display = "none";
+
+        
+        /*
+        Below is ommited for now
+        for (var i = 0; i < profile.projectHistory.length; i++) {
+            const projectElement = document.createElement('li');
+            projectElement.innerText = profile.projectHistory[i];
+            console.log(profile.projectHistory[i]);
+            projectElement.className = 'horiz-center';
+            projectListElement.appendChild(projectElement);
+        }
+        */
+        for (var i = 0; i < profile.reviews.length; i++) {
+            const reviewElement = document.createElement('li');
+            reviewElement.innerText = profile.reviews[i];
+            console.log(profile.reviews[i]);
+            reviewElement.className = 'horiz-center';
+            reviewListElement.appendChild(reviewElement);
+        }
+    });
+
+    // call other loading 
+    loggingIn();
+    loadProjects();
+}
+
+/* displays contact info if logged in. otherwise, asks to log in*/
+function loggingIn(){
+    fetch('authentication').then(response => response.json()).then((authentication) =>{
+        const contactInfoElement = document.getElementById("contact-info");
+        // const logoutElement = document.getElementById("logout");
+        const loggingElement = document.getElementById("logging");
+
+        var isLoggedIn = (authentication.isLoggedIn == 'true'); 
+        var loginUrl = authentication.login;
+        var logoutUrl = authentication.logout;
+        if(isLoggedIn){
+            loggingElement.innerText = "LOG OUT";
+            loggingElement.href = loginUrl;
+            contactInfoElement.display = "inline";
+            // logoutElement.innerHTML = "<p>Logout <a href=\"" + logoutUrl + "\">here</a>.</p>"
+
+        } else {
+            loggingElement.innerText = "LOG IN";
+            loggingElement.href = loginUrl;
+            contactInfoElement.innerHTML = "Login <a href=\"" + loginUrl + "\">here</a> to see the contact info.";
+            contactInfoElement.display = "inline";
+        }
+    });
+}
+
+/* Author: Diego V.A. */
+function loadProjects(){
```

Comment:
I modified this function in my last pull request to separate the fetch part (now with filters) from the parsing (build project cards) part. Maybe you can repurpose it in the future to filter by project author only.